File: src/python/_0345_Reverse_Vowels_of_a_String.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = {"a", "e", "i", "o", "u", "A", "E", "I", "O", "U"}
        s = list(s)
        l, r = 0, len(s) - 1

        while l < r:
            while l < r and s[l] not in vowels:
                l += 1

            while l < r and s[r] not in vowels:
                r -= 1

            s[l], s[r] = s[r], s[l]

            l += 1
            r -= 1

        return "".join(s)

File: src/java/_0345_Reverse_Vowels_of_a_String.java
public class Solution {
    public String reverseVowels(String s) {
        boolean[] vowels = new boolean[128];
        vowels['a'] = true;
        vowels['A'] = true;
        vowels['e'] = true;
        vowels['E'] = true;
        vowels['i'] = true;
        vowels['I'] = true;
        vowels['o'] = true;
        vowels['O'] = true;
        vowels['u'] = true;
        vowels['U'] = true;

        char[] str = s.toCharArray();
        int l = 0, r = s.length() - 1;

        while (l < r) {
            while (l < r && !vowels[str[l]])
                l++;

            while (l < r && !vowels[str[r]])
                r--;

            char temp = str[l];
            str[l] = str[r];
            str[r] = temp;

            l++;
            r--;
        }

        return new String(str);
    }
}
